storage-account.tf

```
@@ -1,5 +1,94 @@
 // Shared and specialised Storage Accounts
 
+locals {
+  mgmt_network_name    = "core-cftptl-intsvc-vnet"
+  mgmt_network_rg_name = "aks-infra-cftptl-intsvc-rg"
+
+  sa_aat_subnets = [
+    data.azurerm_subnet.jenkins_subnet.id,
+    data.azurerm_subnet.aks-00-mgmt.id,
+    data.azurerm_subnet.aks-01-mgmt.id,
+    data.azurerm_subnet.aks-00-infra.id,
+    data.azurerm_subnet.aks-01-infra.id,
+    data.azurerm_subnet.aks-00-preview.id,
+  data.azurerm_subnet.aks-01-preview.id]
+
+  sa_other_subnets = [
+    data.azurerm_subnet.jenkins_subnet.id,
+    data.azurerm_subnet.aks-00-mgmt.id,
+    data.azurerm_subnet.aks-01-mgmt.id,
+    data.azurerm_subnet.aks-00-infra.id,
+  data.azurerm_subnet.aks-01-infra.id]
+
+  sa_subnets = split(",", var.env == "aat" ? join(",", local.sa_aat_subnets) : join(",", local.sa_other_subnets))
+}
+
+data "azurerm_virtual_network" "mgmt_vnet" {
+  provider            = azurerm.mgmt
+  name                = local.mgmt_network_name
+  resource_group_name = local.mgmt_network_rg_name
+}
+
+data "azurerm_subnet" "jenkins_subnet" {
+  provider             = azurerm.mgmt
+  name                 = "iaas"
+  virtual_network_name = data.azurerm_virtual_network.mgmt_vnet.name
+  resource_group_name  = data.azurerm_virtual_network.mgmt_vnet.resource_group_name
+}
+
+data "azurerm_subnet" "aks-00-mgmt" {
+  provider             = azurerm.mgmt
+  name                 = "aks-00"
+  virtual_network_name = data.azurerm_virtual_network.mgmt_vnet.name
+  resource_group_name  = data.azurerm_virtual_network.mgmt_vnet.resource_group_name
+}
+
+data "azurerm_subnet" "aks-01-mgmt" {
+  provider             = azurerm.mgmt
+  name                 = "aks-01"
+  virtual_network_name = data.azurerm_virtual_network.mgmt_vnet.name
+  resource_group_name  = data.azurerm_virtual_network.mgmt_vnet.resource_group_name
+}
+
+data "azurerm_virtual_network" "aks_core_vnet" {
+  provider            = azurerm.aks-infra
+  name                = "core-${var.env}-vnet"
+  resource_group_name = "aks-infra-${var.env}-rg"
+}
+
+data "azurerm_subnet" "aks-00-infra" {
+  provider             = azurerm.aks-infra
+  name                 = "aks-00"
+  virtual_network_name = data.azurerm_virtual_network.aks_core_vnet.name
+  resource_group_name  = data.azurerm_virtual_network.aks_core_vnet.resource_group_name
+}
+
+data "azurerm_subnet" "aks-01-infra" {
+  provider             = azurerm.aks-infra
+  name                 = "aks-01"
+  virtual_network_name = data.azurerm_virtual_network.aks_core_vnet.name
+  resource_group_name  = data.azurerm_virtual_network.aks_core_vnet.resource_group_name
+}
+
+data "azurerm_virtual_network" "aks_preview_vnet" {
+  provider            = azurerm.aks-preview
```

should probably be flagged for only running in aat
```suggestion
  provider            = azurerm.aks-preview
  count               = var.env == "aat" ? 1 : 0
```